tests/MongoDB.Analyzer.Tests.Common.TestCases/Linq/LinqQuery.cs

```diff
@@ -0,0 +1,128 @@
+                        where item.Age == 22 && (item.LastName == "Doe" || item.Name == "John")
+                        select item.Address;
+        }
+
+        [MQL("aggregate([{ \"$match\" : { \"$and\" : [{ \"$or\" : [{ \"Age\" : 22 }, { \"Age\" : 25 }] }, { \"$or\" : [{ \"LastName\" : \"Doe\" }, { \"Name\" : \"John\" }] }] } }, { \"$project\" : { \"Age\" : \"$Age\", \"_id\" : 0 } }])")]
+        public void Query_Syntax_5()
+        {
+            var query = from item in GetMongoQueryable()
+                        where (item.Age == 22 || item.Age == 25) && (item.LastName == "Doe" || item.Name == "John")
+                        select item.Age;
+        }
+
+        [MQL("aggregate([{ \"$match\" : { \"$and\" : [{ \"$or\" : [{ \"Age\" : 22 }, { \"Age\" : 25 }] }, { \"$or\" : [{ \"LastName\" : \"Doe\" }, { \"Name\" : \"John\" }] }] } }, { \"$project\" : { \"Age\" : \"$Age\", \"Address\" : \"$Address\", \"_id\" : 0 } }])")]
+        public void Query_Syntax_6()
+        {
+            var query = from item in GetMongoQueryable()
+                        where (item.Age == 22 || item.Age == 25) && (item.LastName == "Doe" || item.Name == "John")
+                        select new { item.Age, item.Address };
+        }
+
+        [MQL("aggregate([{ \"$match\" : { \"$or\" : [{ \"Age\" : 22 }, { \"Age\" : 25 }] } }, { \"$match\" : { \"$or\" : [{ \"LastName\" : \"Doe\" }, { \"Name\" : \"John\" }] } }, { \"$project\" : { \"Age\" : \"$Age\", \"Address\" : \"$Address\", \"_id\" : 0 } }])")]
+        public void Query_Syntax_7()
+        {
+            var query = from item in GetMongoQueryable()
+                        where (item.Age == 22 || item.Age == 25)
+                        where (item.LastName == "Doe" || item.Name == "John")
+                        select new { item.Age, item.Address };
+        }
+
+        [MQL("aggregate([{ \"$match\" : { \"$or\" : [{ \"Age\" : 22 }, { \"Age\" : 25 }] } }, { \"$match\" : { \"$or\" : [{ \"LastName\" : \"Doe\" }, { \"Name\" : \"John\" }] } }, { \"$match\" : { \"Address\" : /Drive$/s } }, { \"$project\" : { \"Age\" : \"$Age\", \"Address\" : \"$Address\", \"_id\" : 0 } }])")]
+        public void Query_Syntax_8()
+        {
+            var query = from item in GetMongoQueryable()
+                        where (item.Age == 22 || item.Age == 25)
+                        where (item.LastName == "Doe" || item.Name == "John")
+                        where item.Address.EndsWith("Drive")
+                        select new { item.Age, item.Address };
+        }
+
+        [MQL("aggregate([{ \"$match\" : { \"$or\" : [{ \"Age\" : 22 }, { \"Age\" : 25 }] } }, { \"$match\" : { \"$or\" : [{ \"LastName\" : \"Doe\" }, { \"Name\" : \"John\" }] } }, { \"$match\" : { \"Address\" : /^Drive/s } }, { \"$project\" : { \"Age\" : \"$Age\", \"Address\" : \"$Address\", \"_id\" : 0 } }])")]
+        public void Query_Syntax_9()
+        {
+            var query = from item in GetMongoQueryable()
```

Comment:
Try having large difference between test cases, reflected in the test method name, for example `Query_Syntax_8` and `Query_Syntax_9` are practically identical.